### PR TITLE
[MODULAR] Lazily adds trash piles to TG maps.

### PR DESCRIPTION
**Pull request by YakumoChen**
## About The Pull Request
I attempted to up port trash piles to TG to put them on all maps (https://github.com/tgstation/tgstation/pull/59393) It was closed, but I'm using a solution to add them to maps without mapping  I used on that PR so I cam have them here
This replaces 1 in 3 random loot crate spawns with trash pile structures instead.

## Why It's Good For The Game
Maint loot is fun. All of maint having no loot within 5 minutes of round start is not fun.

## Changelog
:cl:
qol: trash piles now spawn on all maps
/:cl:

### CODE REVIEW

**Comment by YakumoChen** [comment]
This errant changelog shit does not go away does it
I'm 3 hours away from a real computer so hold on

**Comment by Funce** [comment]
This is a hack and a half, and I don't like this.
If there's a way to adjust this "problem" (muh loots???) then I'd like to see it done another way.

Also, disable the actions on your fork to stop the changelog compiles.

**Comment by YakumoChen** [comment]
@Funce how is this a hack? It takes advantage of an existing system that shares the same purpose. Trash piles are already in the game and extensively used on Journey. You can read my TG PR for why it's a good feature.

**Comment by Funce** [comment]
This is actually a 1000 weighting, and I was appalled at the thought of a 500% weighting for a container, not even reading what it was before. The chance change here is fine.

But this can be modularised though. Override the loot table in a new file on modular.
```dm
/obj/effect/loot_site_spawner
	///This is the loot table for the spawner. Try to make sure the weights add up to 1000, so it is easy to understand.
	var/list/loot_table = list(/obj/structure/closet/crate/maint = 510,
							/obj/structure/trash_pile = 255,
							/obj/structure/closet/crate/trashcart/filled = 75,
							/obj/effect/spawner/bundle/moisture_trap = 50,
							/obj/effect/spawner/bundle/hobo_squat = 30,
							/obj/structure/closet/mini_fridge = 35,
							/obj/effect/spawner/lootdrop/gross_decal_spawner = 30,
							/obj/structure/closet/crate/decorations = 15)
```

**Comment by Gandalf2k15** [comment]
Do these actually spawn mobs?

**Comment by YakumoChen** [comment]
@Funce I'm not familiar with how to override. Do I just put an identical the file in ~skyrat_defines?

**Comment by YakumoChen** [comment]
> 
> 
> Do these actually spawn mobs?

Not that I believe? This doesn't change the behavior of anything currently already in the game.

**Comment by YakumoChen** [comment]
I'm pretty sure I did what Funce asked me to do, but I'm not entirely sure. If it works, it works, I hope.

**Comment by Funce** [comment]
You'll only need to include what's in my reply for inside this new file.

Can you move the file into a similar folder as the original it overrides? `modular_skyrat/master_files/code/game/objects/effects/spawners/`?

File: modular_skyrat/master_files/code/game/objects/effects/spawners/lootsite.dm [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/obj/effect/loot_site_spawner
+	///This is the loot table for the spawner. Try to make sure the weights add up to 1000, so it is easy to understand.
+	var/list/loot_table = list(/obj/structure/closet/crate/maint = 510,
```
Comment (Funce) [inline modular_skyrat/master_files/code/game/objects/effects/spawners/lootsite.dm]:
```suggestion
	loot_table = list(/obj/structure/closet/crate/maint = 510,
```

**Pull request merged**